### PR TITLE
Add ci test report coverage and upload to codecov

### PR DESCRIPTION
**Pull request by yaalsn**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -257,7 +259,8 @@
                 </executions>
                 <configuration>
                     <generateGitPropertiesFile>true</generateGitPropertiesFile>
-                    <generateGitPropertiesFilename>${project.build.outputDirectory}/git.properties</generateGitPropertiesFilename>
+                    <generateGitPropertiesFilename>${project.build.outputDirectory}/git.properties
```
Comment (timmyyuan) [inline pom.xml:262]: Can you format this xml file ? Thanks

File: pom.xml [hunk 2]
```diff
@@ -268,6 +271,31 @@
                 </configuration>
             </plugin>
 
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
```
Comment (timmyyuan) [inline pom.xml:276]: Could you explain the benefits of using jacoco maven plugin?

**Review by timmyyuan — COMMENTED** [review]
Can you show us the result ? e.g., how does this tool report test coverage (you post some screenshots or reports here)

File: pom.xml (hunk 1 above)
Comment (yaalsn) [reply pom.xml]: Ok~done

File: pom.xml (hunk 2 above)
Comment (yaalsn) [reply pom.xml:277]: - The most used and the community is active

- Has enough [functions](https://openclover.org/doc/manual/4.2.0/general--comparison-of-code-coverage-tools.html#Comparisonofcodecoveragetools-legend) to maintain `mop` project

- I tried other tool like `cobertura` but it parses [lambda expression error](https://github.com/streamnative/mop/runs/4296458283?check_suite_focus=true)
image . The same issues are https://github.com/cobertura/cobertura/issues/176 https://github.com/stripe/stripe-java/pull/891

**Comment by yaalsn** [comment]
> Can you show us the result ? e.g., how does this tool report test coverage (you post some screenshots or reports here)

Beisdes the robot(`codecov-commenter`) report the coverage of commit above, the detail can be viewed in [codecov website](https://app.codecov.io/gh/streamnative/mop/blob/2021_11_22_add_ci_test_report/mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/TopicFilterImpl.java).

**Comment by timmyyuan** [comment]
@yaalsn Thanks a lot for your investigation. I think this is a nice job and we can add a badge (with the permissions of streamnative) in the future.

**Comment by Technoboy-** [comment]
@timmyyuan 
Could we need to merge this pr ?

**Comment by timmyyuan** [comment]
@Technoboy- The PR failed to pass mop mvn build check. I think we can merge it after the check passes.

**Comment by yaalsn** [comment]
Test coverage tool mainly uses JVMTI Agent and It will increase the running time of every test case. We need to resolve the problem of test case timeout before merge the PR.

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**